vault/resource_pki_secret_backend_config_auto_tidy.go

```diff
@@ -0,0 +1,234 @@
+func pkiSecretBackendConfigAutoTidySuppressDurationDiff(key, oldValue, newValue string, _ *schema.ResourceData) bool {
+	if _, isDuration := pkiSecretBackendConfigAutoTidyDurationFields[key]; !isDuration {
+		return false
+	}
+	// The old value is what is returned by auto-tidy config. It will be either the empty
+	// string or the number of seconds. In the case of pause_duration which does return
+	// a duration string, Atoi() will fail making this function return false, which is
+	// the correct thing to do since we don't want to suppress the diff in that case.
+	seconds, err := strconv.Atoi(oldValue)
+	if err != nil {
+		return false
+	}
+	// The new value is what we have in the state, which will be a duration string.
+	duration, err := time.ParseDuration(newValue)
+	if err != nil {
+		return false
+	}
+	return seconds == int(duration.Seconds())
+}
+
+func pkiSecretBackendConfigAutoTidyCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+
+	path := pkiSecretBackendConfigAutoTidyPath(backend)
+
+	action := "Create"
+	if !d.IsNewResource() {
+		action = "Update"
+	}
+
+	data := map[string]interface{}{}
+
+	for k, s := range pkiSecretBackendConfigAutoTidySchema() {
+		if k == consts.FieldBackend {
+			continue
+		}
+		if s.Type == schema.TypeBool {
+			data[k] = d.Get(k)
+		} else { // all other fields are duration strings
+			if v, ok := d.GetOk(k); ok {
+				data[k] = v
+			}
+		}
+	}
+
+	log.Printf("[DEBUG] %s auto tidy config on PKI secret backend %q", action, backend)
+	_, err := client.Logical().Write(path, data)
+	if err != nil {
+		return diag.Errorf("error writing PKI auto tidy config to %q: %s", backend, err)
+	}
+	log.Printf("[DEBUG] %sd auto tidy config on PKI secret backend %q", action, backend)
+
+	if d.IsNewResource() {
+		d.SetId(fmt.Sprintf("%s/config/auto-tidy", backend))
+	}
+
+	return pkiSecretBackendConfigAutoTidyRead(ctx, d, meta)
+}
+
+func pkiSecretBackendConfigAutoTidyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	path := d.Id()
+
+	if path == "" {
+		return diag.Errorf("no path set, id=%q", d.Id())
+	}
+
+	backend, found := strings.CutSuffix(path, "/config/auto-tidy")
+	if !found {
+		return diag.Errorf("error parsing backend from ID %s", path)
+	}
+	if err := d.Set(consts.FieldBackend, backend); err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] Reading auto tidy config from PKI secret path %q", path)
+	config, err := client.Logical().Read(path)
+	if err != nil {
+		return diag.Errorf("error reading auto tidy config on PKI secret backend %q: %s", path, err)
+	}
+
+	if config == nil {
+		log.Printf("[WARN] Removing auto tidy config path %q as its ID is invalid", path)
```

Comment:
Should this not be an error, if we received a nil response from the server something went wrong I believe?

Reply:
Good catch, I have changed this to return an error as you suggest.